azerothcore/azerothcore-wotlk
# feat(Core): Implement world availability option

## Changes Proposed:
-  Implement realm availability enable / disable toggle

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8751

## SOURCE:

Port from Vanilla Mangos https://github.com/vmangos/core/commit/908179f75f78326de36ec70fd207b341d1c20632

## Tests Performed:

- Tested with Ubuntu 21.04

## How to Test the Changes:

1. Disable availability option in worldserver.conf from `1` to `0`
2. Start your server
3. Login to an account
4. Try to login selected realm (with current PR, should give you a `World Down` message. Can be changed to what ever)


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
